lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1145,6 +1116,11 @@ def run(self):
         return msg, changed, rebuilt, differences.get_legacy_docker_diffs(), facts
 
 
+def _detect_publish_mode_usage(client):
+    for publish_def in client.module.params['publish']:
+        return bool(publish_def.get('mode'))
```

Comment:
Hmm, actually, that won't work. In case the first entry does not use `mode`, but the second does, this already returns `False`. I think the function should look as follows:
```.py
def _detect_publish_mode_usage(client):
    for publish_def in client.module.params['publish']:
        if publish_def.get('mode'):
            return True
    return False
```
Sorry for not noticing that earlier...

Reply:
Good catch.